catalogd/internal/storage/localdir.go

```diff
@@ -20,95 +26,266 @@ import (
 // done so that clients accessing the content stored in RootDir/catalogName have
 // atomic view of the content for a catalog.
 type LocalDirV1 struct {
-	RootDir string
-	RootURL *url.URL
+	RootDir            string
+	RootURL            *url.URL
+	EnableQueryHandler bool
+
+	m  sync.RWMutex
+	sf singleflight.Group
 }
 
-const (
-	v1ApiPath = "api/v1"
-	v1ApiData = "all"
+var (
+	_                Instance = &LocalDirV1{}
+	ErrInvalidParams          = errors.New("invalid parameters")
```

Comment:
Since this is only used inside this package, it should be unexported:
```suggestion
	errInvalidParams          = errors.New("invalid parameters")
```